### DISCUSSION TITLE
Automatically add policy on CreateBucket

### DISCUSSION THREAD

**Discussion by MrVibe** [supabase/supabase#26322] [Questions]
Since supabase supports CreateBucket , is there any way we can add policy on new buckets ? I hae sweeping policies but it seems they're not working on new buckets created by service_role user.

**Comment by Wellusioner** [comment]
I am also looking for a way to do so

**Comment by MrVibe** [comment]
I guess the only solution here is to use Service role key protect it via api.

**Comment by GaryAustin1** [comment]
If you have generic bucket policies that don't depend on the bucket name (or use a partial string you can match for your programmatically created buckets) then it should apply to new buckets also.

**Comment by MrVibe** [comment]
This is what we get in current configuration : https://prnt.sc/FTWzmKN1ruOo

"Error uploading abc.png new row violates row-level security policy"

what could be missing ?

**Reply by GaryAustin1** [reply]
Uploading is not based on bucket policy you show.  The storage.buckets policy is on actually creating new buckets.  You don't normally want users to do that.

Uploading files RLS is based on storage.objects table.
